readthedocs/api/v3/serializers.py

```diff
@@ -443,6 +470,15 @@ def get_versions(self, obj):
         path = reverse("project_version_list", kwargs={"project_slug": obj.slug})
         return self._absolute_url(path)
 
+    def get_downloads(self, obj):
+        path = reverse("project_downloads", kwargs={"project_slug": obj.slug})
+        return self._absolute_url(path)
+
+    def get_documentation(self, obj):
+        version_slug = getattr(self.parent, "version_slug", None)
+        resolver = getattr(self.parent, "resolver", Resolver())
+        return obj.get_docs_url(version_slug=version_slug, resolver=resolver)
```

Comment:
If we already have the version object, we can just call `resolver.resolve_version`, so it doesn't query the version object from the DB.

Reply:
My two guesses:

- Maybe the version doesn't have all fields (e.g, "only" was used).
- A method needs to be cached.